### DISCUSSION TITLE
What font is used in the default system menu bar?

### DISCUSSION THREAD

**Discussion by es183923** [FelixKratz/SketchyBar#187] [Q&A] [answered]
Here is a video:

https://user-images.githubusercontent.com/72744903/168888658-1cee6f99-8725-48e0-a05c-7dcfe21b0375.mov

See how the text doesn't move, and only one character moves at a time.

This font is not monospace looking, but it acts as monospace. I was wondering what font this was, so I could use it as my font too. (I realize this is kinda off-topic, but I don't know any better place to ask)

**Comment by FelixKratz** [comment, marked as answer]
I believe this is the San Francisco font family. In my bar I use the SF Pro font, which looks very similar: https://developer.apple.com/fonts/

**Reply by es183923** [reply]
It could be, but that font isn't monspaced. SF Mono is a standard monospace font, but it is the standard typewriter-style font, not a sans serif font.

**Reply by FelixKratz** [reply]
Another hint: https://developer.apple.com/design/human-interface-guidelines/macos/visual-design/typography/

When I run
```objc
NSFont* font = [NSFont menuBarFontOfSize:16];
NSLog(@"%@",[font fontName]);
```
it prints ".AppleSystemUIFont", which should be some San Francisco Variant. I you want you can look at `/System/Library/Fonts/` and see all system fonts there.

**Reply by liangch** [reply]
I guess the open font feature **tnum** is used.
FYI. https://rsms.me/inter/#features/tnum

**Reply by es183923** [reply]
Yeah, that looks about right. Is it possible for tnum to be added to sketchybar somehow? Is it possible for the tnum feature to be implemented in sketchybar?

**Reply by FelixKratz** [reply]
Sure that would be possible, we would need to add the font attribute `kCTFontFeatureSettingsAttribute` and in the dictionary select the feature `kMonospacedNumbersSelector` when creating the font descriptor here: https://github.com/FelixKratz/SketchyBar/blob/c5a9f96cda5bea11bb82205dfd1f070b2cc4734e/src/text.c#L34 

We would have to discuss about how the actual `font="..."` syntax would need to be adapted for this, because there are many features and I would want to add more than this single one while doing the changes (e.g. ligatures).

**Reply by FelixKratz** [reply]
Further reading: https://developer.apple.com/fonts/TrueType-Reference-Manual/RM09/AppendixF.html, https://developer.apple.com/documentation/coretext/kctfontfeaturesettingsattribute for the implementation. If you want you can create an enhancement issue for this where we can condense all of this information and when I have some free time I will look if I can get it working, or someone creates a pr for this.

**Reply by liangch** [reply]
Found a workaround, have fun.

1. find the font you like and with open type feature to make num looks like mono.  I use **Inter** font[1], which has open type feature **tnum**.
2. use **OpenType Feature Freezer**[2] to create a font which has this font feature enabled by default. Below command makes a new font InterT which has **tnum** enabled by default. -R means replacing name Inter in font with InterT.
	  ```
	  pyftfeatfreeze -f 'tnum' -R 'Inter/InterT' Inter.ttf InterT.ttf
	  ```
3. use this new font in sketchybar (e.g. clock item).

Refereces:
[1]. Inter font https://rsms.me/inter/
[2]. https://github.com/twardoch/fonttools-opentype-feature-freezer
